grails-app/controllers/org/broadinstitute/orsp/UserController.groovy

```diff
@@ -29,4 +37,53 @@ class UserController {
         render syncedUsers as JSON
     }
 
+    def rolesManagement() {
+        render(view: "/rolesManagement/index")
+    }
+
+    def getOrspUsers() {
+        JSON.registerObjectMarshaller(User) {
+            def output = [:]
+            output['id'] = it.id
+            output['userName'] = it.userName
+            output['displayName'] = it.displayName
+            output['emailAddress'] = it.emailAddress
+            output['createdDate'] = it.createdDate
+            output['updatedDate'] = it.updatedDate
+            output['lastLoginDate'] = it.lastLoginDate
+            output['roles'] = it.roles.collect{r -> r.getRole()}
+            return output
+        }
+
+        PaginationParams pagination = new PaginationParams(
+                draw: params.getInt("draw") ?: 1,
+                start: params.getInt("start")?: 0,
+                length: params.getInt("length")?: 10,
+                orderColumn: params.getInt("orderColumn")? params.get("orderColumn"): 0,
+                sortDirection: params.get("sortDirection")? params.get("sortDirection").toString() : "asc",
+                searchValue: params.get("searchValue")? params.get("searchValue").toString() : null)
+        render(queryService.queryUserRoles(pagination) as JSON)
+    }
+
+    def editOrspUserRole() {
+        Integer userId = request.JSON["userId"] as Integer
+        ArrayList<String> rolesToAssign = request.JSON["roles"] as ArrayList<String>
+
+        try {
+            userService.editUserRoles(userId, rolesToAssign)
+            response.status = 200
+            render([message: 'Role Updated'] as JSON)
+        } catch(IllegalArgumentException e) {
+            response.status = 400
+            render([error: e.message] as JSON)
+        } catch(SQLException e) {
+            response.status = 500
+            log.error("Error trying to execute delete/insert for new roles: ${rolesToAssign} to userId: ${userId}. " + e.message)
+            render([error: e.message] as JSON)
```

Comment:
I suggest we don't return the full error message here. Better to log it and then send the user something generic so that raw sql doesn't go back to the user.

Reply:
This is now changed. Replaced this render error message to `"An error has ocurred trying to update roles"`